graphql/schemabuilder/pagination.go

```
@@ -433,65 +433,103 @@ func (c *connectionContext) applyBatchTextFilter(ctx context.Context, nodes []in
 	return nodesToKeep, nil
 }
 
+func (c *connectionContext) checkFilters(ctx context.Context, node interface{}, matchStrings []string, filterFields map[string]*graphql.Field) (bool, error) {
+	keep := false
+	for name, filterField := range filterFields {
+		// Resolve the graphql.Field made for sorting.
+		text, err := filterField.Resolve(ctx, node, nil, nil)
+		if err != nil {
+			return keep, err
+		}
+		// Only strings are allowed for FilterText fields.
+		textString, ok := text.(string)
+		if !ok {
+			return keep, fmt.Errorf("filter %s returned %T, must be a string", name, text)
+		}
+		if filter.MatchText(textString, matchStrings) {
+			keep = true
+			break
+		}
+	}
+	return keep, nil
+}
+
+func (c *connectionContext) applyTextFilterNotBatchedExpensive(ctx context.Context, nodes []interface{}, matchStrings []string, filterFields map[string]*graphql.Field) ([]bool, error) {
+	g, ctx := errgroup.WithContext(ctx)
+	nodesToKeep := make([]bool, len(nodes))
+	if len(filterFields) > 0 {
+		for unscopedI, unscopedNode := range nodes {
+			i, node := unscopedI, unscopedNode
+			g.Go(func() error {
+				keep, err := c.checkFilters(ctx, node, matchStrings, filterFields)
+				nodesToKeep[i] = keep
+				return err
+			})
+		}
+	}
+	if err := g.Wait(); err != nil {
+		return nil, err
+	}
+	return nodesToKeep, nil
+}
+
+func (c *connectionContext) applyTextFilterNotBatched(ctx context.Context, nodes []interface{}, matchStrings []string, filterFields map[string]*graphql.Field) ([]bool, error) {
+	nodesToKeep := make([]bool, len(nodes))
+	if len(filterFields) > 0 {
+		for unscopedI, unscopedNode := range nodes {
+			i, node := unscopedI, unscopedNode
+			keep, _ := c.checkFilters(ctx, node, matchStrings, filterFields)
```

should we propagate any errors to the caller here?